src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -110,14 +127,29 @@ private static double GetFragmentation()
         }
 #endif
 
-        private IEnumerable<Measurement<double>> GetProcessorTimes()
+#if NET6_0_OR_GREATER
+        private static IEnumerable<Measurement<long>> GetGarbageCollectionHeapSizes()
         {
-            var process = Process.GetCurrentProcess();
-            return new[]
+            var generationInfo = GC.GetGCMemoryInfo().GenerationInfo;
+
+            // TODO: Confirm 1. that the number of heaps for garbage collection will not change, at least during the lifetime of a process.
+            // and 2. that there will not be more than 5 heaps, at least for the existing .NET version that is supported (net6.0).
+            Debug.Assert(generationInfo.Length <= HeapNames.Length, "There should not be more than 5 heaps");
```

Comment:
```suggestion
            Debug.Assert(generationInfo.Length <= HeapNames.Length, "There should not be more than 5 generations");
```

Reply:
I thought saying "generation" here might be confusing as there are 3 generations (generation 0, generation 1, generation 2), while there are 5 segments of heaps: generation 0, generation 1, generation 2, large object heap (which is sometimes referred to as generation 3) and [Pinned Object Heap](https://devblogs.microsoft.com/dotnet/internals-of-the-poh/): https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/fundamentals